src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleExtensionMetadata.java

```diff
@@ -311,73 +299,52 @@ private static Optional<RootModuleFileFixup> generateFixup(
 
     message += "Fix the use_repo calls by running 'bazel mod tidy'.";
 
-    var buildozerCommands =
-        Stream.of(
-                makeUseRepoCommand(
-                    "use_repo_add",
-                    false,
-                    importsToAdd,
-                    extensionBzlFile,
-                    extensionName,
-                    rootUsage.getIsolationKey()),
-                makeUseRepoCommand(
-                    "use_repo_remove",
-                    false,
-                    importsToRemove,
-                    extensionBzlFile,
-                    extensionName,
-                    rootUsage.getIsolationKey()),
-                makeUseRepoCommand(
-                    "use_repo_add",
-                    true,
-                    devImportsToAdd,
-                    extensionBzlFile,
-                    extensionName,
-                    rootUsage.getIsolationKey()),
-                makeUseRepoCommand(
-                    "use_repo_remove",
-                    true,
-                    devImportsToRemove,
-                    extensionBzlFile,
-                    extensionName,
-                    rootUsage.getIsolationKey()))
-            .flatMap(Optional::stream)
-            .collect(toImmutableList());
+    var moduleFilePathToCommandsBuilder = ImmutableListMultimap.<PathFragment, String>builder();
+    // Repos to add are easy: always add them to the first proxy of the correct type.
+    if (!importsToAdd.isEmpty()) {
+      Proxy firstNonDevProxy =
+          rootUsage.getProxies().stream().filter(p -> !p.isDevDependency()).findFirst().get();
+      moduleFilePathToCommandsBuilder.put(
+          firstNonDevProxy.getContainingModuleFilePath(),
+          makeUseRepoCommand("use_repo_add", firstNonDevProxy.getProxyName(), importsToAdd));
```

Comment:
the proxy name would be the empty string, resulting in commands like `use_repo_add  repo1 repo2`, which means `repo1` would be interpreted as the exported name...

I changed it so that unexported proxies would be treated as if they're exported as `_unnamed_usage`. Will probably tip people off that these are supposed to be named.